core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3077,10 +3077,11 @@ private boolean syncMetadata(JournalContext journalContext, LockedInodePath inod
     // The high-level process for the syncing is:
     // 1. Find all Alluxio paths which are not consistent with the corresponding UFS path.
     //    This means the UFS path does not exist, or is different from the Alluxio metadata.
-    // 2. Delete those Alluxio paths which are not consistent with Alluxio. After this step, all
+    // 2. If a directory is not consistent, update the directory inode permissions from UFS.
```

Comment:
Also include what do we do if the directory exists in Alluxio but not in UFS?

Reply:
I updated the comment. If a path does not exist in UFS, it will be deleted.